web/src/server_events_dispatch.js

```diff
@@ -546,20 +546,22 @@ export function dispatch_normal_event(event) {
                         const is_narrowed_to_stream = narrow_state.is_for_stream_id(
                             stream.stream_id,
                         );
+                        stream_data.delete_sub(stream.stream_id);
                         if (is_narrowed_to_stream) {
                             message_lists.current.update_trailing_bookend();
                         }
-                        stream_data.delete_sub(stream.stream_id);
                         stream_settings_ui.remove_stream(stream.stream_id);
+                        message_view_header.maybe_rerender_title_area_for_stream(stream);
+                        message_live_update.rerender_messages_view();
```

Comment:
Can this line be pullet out of the loop? It's a very expensive operation.

Reply:
Done.